src/Codeception/Lib/Console/DiffFactory.php

```diff
@@ -9,26 +12,17 @@
  **/
 class DiffFactory
 {
-    /**
-     * @param ComparisonFailure $failure
-     * @return string|null
-     */
-    public function createDiff(ComparisonFailure $failure)
+    public function createDiff(ComparisonFailure $failure): ?string
     {
         $diff = $this->getDiff($failure->getExpectedAsString(), $failure->getActualAsString());
-        if (!$diff) {
+        if ($diff === '') {
```

Comment:
`createDiff` method is called in a [single place](https://github.com/Codeception/Codeception/pull/6129/files#diff-d3024ad9f62f7f2a33ee201f9b31494397d8d18c0936c30f49275dd673ff4e7bL49) which could be updated to handle empty string response and simplify return type to `: string`.